quality-gate.sh

```diff
@@ -39,6 +39,24 @@ mli-files () {
   fi
 }
 
+structural-equality () {
+  # prefer =,<> to  ==,!=
+  if git grep -r --count ' == ' -- '**/*.ml' ':!ocaml/sdk-gen/**/*.ml'; then
+    echo "ERROR expected no usages of ' == '" 1>&2
+    exit 1
+  else
+    echo "OK found no usages of ' == '"
+  fi
+
+  if git grep -r --count ' != ' -- '**/*.ml' ':!ocaml/sdk-gen/**/*.ml'; then
+    echo "ERROR expected no usages of ' != '" 1>&2
```

Comment:
Could we suggest using <> in the message?, it's not clear what the action should be taken unless the user knows the difference in equalities.